File: LeetCode/Arrays & Hashing/find-pivot-index.py
class Solution(object):
    def pivotIndex(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        total = sum(nums)
        left_sum = 0
        for index,_ in enumerate(nums):
            total -= nums[index]
            if total == left_sum:
                return index
            else:
                left_sum  += nums[index]
        return -1


s = Solution()
print(s.pivotIndex([1,7,3,6,5,6]))

File: LeetCode/Arrays & Hashing/find-pivot-index.js
var arrSum = function(numsArr){
    var totalSum = 0;
    for(var value of numsArr){
        totalSum += value;
    }
    return totalSum;
}


var pivotIndex = function(nums) {
    let total = arrSum(nums);
    let leftSum = 0;
    for(let index in nums){
        total -= nums[index]
        if(total == leftSum){
            return index
        }else{
            leftSum += nums[index]
        }
    }
    return -1

};

console.log(pivotIndex([1,7,3,6,5,6]))